allure-notifications-api/src/test/java/guru/qa/allure/notifications/clients/NotificationTests.java

```diff
@@ -0,0 +1,126 @@
+package guru.qa.allure.notifications.clients;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mockConstruction;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.MockedConstruction;
+import org.mockito.MockedStatic;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import guru.qa.allure.notifications.chart.Chart;
+import guru.qa.allure.notifications.config.Config;
+import guru.qa.allure.notifications.config.base.Base;
+import guru.qa.allure.notifications.config.enums.Language;
+import guru.qa.allure.notifications.exceptions.MessagingException;
+import guru.qa.allure.notifications.json.JSON;
+import guru.qa.allure.notifications.model.legend.Legend;
+import guru.qa.allure.notifications.model.phrases.Phrases;
+import guru.qa.allure.notifications.model.summary.Summary;
+import guru.qa.allure.notifications.template.data.MessageData;
+
+@ExtendWith(MockitoExtension.class)
+class NotificationTests {
+    private static final String PHRASES_PATH = "/phrases/en.json";
+    private static final String LEGEND_PATH = "/legend/en.json";
+    private static final String ALLURE_PATH = "/allure";
+
+    private static final Config CONFIG = new Config();
+    private static final Base BASE = new Base();
+    private static final Summary SUMMARY = new Summary();
+    private static final Phrases PHRASES = new Phrases();
+
+    @Mock private Notifier notifier;
+    @Mock private Path path;
+
+    @BeforeEach
+    void init()
+    {
+        BASE.setAllureFolder(ALLURE_PATH);
+        BASE.setLanguage(Language.en);
+        CONFIG.setBase(BASE);
+    }
+
+    static Stream<Arguments> inputData() {
+        return Stream.of(
+                Arguments.of(false, null,         0, 1),
+                Arguments.of(true,  new Legend(), 1, 0)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputData")
+    void shouldSendNotification(boolean chartEnabled, Legend legend, int chartActionsCount, int textActionsCount)
+            throws MessagingException, IOException {
+        BASE.setEnableChart(chartEnabled);
```

Comment:
please avoid mutation of static data in tests (it's better to use instance fields or local variables)